test/integration2/dialects/mssql.spec.js

```diff
@@ -0,0 +1,179 @@
+            const constraintName = 'DF_test_name';
+            await knex.schema.alterTable('test', function () {
+              this.string('name').defaultTo('knex', { constraintName });
+            });
+            const { name } = await fetchDefaultConstraint(knex, 'test', 'name');
+            expect(name).to.equal('DF_test_name');
+          });
+          it("doesn't name default constraint", async () => {
+            const constraintName = '';
+            await knex.schema.alterTable('test', function () {
+              this.string('name').defaultTo('knex', { constraintName });
+            });
+            const { name } = await fetchDefaultConstraint(knex, 'test', 'name');
+            // this is the default patten used by mssql if no constraint is defined
+            expect(name).to.match(/^DF__test__name__[0-9A-Z]+$/);
+          });
+        });
+
+        describe('comment limits', () => {
+          // NOTE: We are using varchar, not nvarchar, so we can hit an odd number of bytes with our characters.
+          const byteCount0 = "''";
+          const byteCount7500 = "'" + 'X'.repeat(7500) + "'";
+          const byteCountOver7500 = "'" + 'X'.repeat(7501) + "'";
+
+          const tableTarget = { schemaName: 'dbo', tableName: 'test' };
+          it('supports table comments of max 7500 bytes', async () => {
+            // 0 works
+            await expect(comment(knex, 'add', byteCount0, tableTarget)).to
+              .eventually.be.fulfilled;
+
+            // 7500 works
+            await expect(comment(knex, 'update', byteCount7500, tableTarget)).to
+              .eventually.be.fulfilled;
+
+            // Over 7500 fails
+            await expect(
+              comment(knex, 'update', byteCountOver7500, tableTarget)
+            ).to.eventually.be.rejectedWith(
+              'The size associated with an extended property cannot be more than 7,500 bytes.'
+            );
+          });
+
+          const columnTarget = { ...tableTarget, columnName: 'id' };
+          it('supports column comments of max 7500 bytes', async () => {
+            // 0 works
+            await expect(comment(knex, 'add', byteCount0, columnTarget)).to
+              .eventually.be.fulfilled;
+
+            // 7500 works
+            await expect(comment(knex, 'update', byteCount7500, columnTarget))
+              .to.eventually.be.fulfilled;
+
+            // Over 7500 fails
+            await expect(
+              comment(knex, 'update', byteCountOver7500, columnTarget)
+            ).to.eventually.be.rejectedWith(
+              'The size associated with an extended property cannot be more than 7,500 bytes.'
+            );
+          });
+
+          // Characters in the supplementary plane need 4 bytes in UTF-16.
+          // (They also happen to need 4 in UTF-8, so this estimate is a useful worst-case for a UTF-8 collation, as well.)
+          const N = 7500 / 4;
```

Comment:
❗ i'm not confident our estimation here based on string length is right, though, because JS will treat the astral plane character i'm using as having a length of 2. we can probably instead allow the string.length to be 7500/2 as a result!

**TODO:** confirm this by asserting against the test comment raw string length, & revise the estimate if it proves out.